applications/blockchainia.md

```diff
@@ -0,0 +1,139 @@
+ - **Costs:** $25,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | Apache 2.0 / GPLv3 / MIT / Unlicense |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can (for example) spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. Unit tests will be written for relevent game server functionality. Unit tests will be written for the DeathToll pallet. In each guide (in each project's Github repo), we will describe how to run these tests. Integration testing will be enabled through game functionality (i.e, Given an enemy is eliminated on server, the proper state functions are executed on chain, and the leaderboard updates through the in-game chain browser) |
+| **0d.** | Docker | We will provide Dockerfiles that can be used to run and test all the functionality delivered with this milestone. This will include a deployable "Full Node" that encases both a game server and substrate node. These will interact with our game client and substrate parachain respectively. A successful MVP will allow a game client to register events on the game server (Server-Authoritative model), which is then written to the parachain ledger. Upon completion of the initial game "level", an "accolade" NFT will be deposited via ink! smart contract to the wallet that completed the required in-game tasks. |
+| 1. | Social Media | We will deploy a social media marketing campaign upon the successful development and testing of an MVP. This will include a social media campaign, building a discord community of evangelists, and marketing materials that describe our project to potential stakeholders and various customers of our ecosystem. The first milestone will involve hitting 1000 followers, which will enable us to involve our community from Milestone 2 forward. 
+| 2. | Substrate module: DeathToll Game Events | The DeathToll Game Events Substrate module will add functionality for starting/ending a "match"", start and end a "round", eliminate a player, score a point, complete a "task". Tasks examples include "move through a defined space", eliminate an AI enemy, retrieve an item, etc. This deliverable will include unit tests and instructions in the git repository. It is highly likely that we will break this into two pallets, one for core gaming functionality, and one specific to the needs of the DeathToll game-type. 
+| 3. | Unity Game/Server | Both DeathToll Modules of our custom chain will interact with with open-source scrips we create to control our Unity created game server.  Each of these processes will initially need to be deployable on the same machine via docker container, until appropriate configuration is added. 
```

Comment:
Is the whole server going to be open source/reproducible, or what is being delivered here?

Reply:
To add on @semuelle's question: We don't support deployment efforts through the W3F grants program.

Reply:
The required pallets and library of scripts that control the interactions occurring between game server and blockchain node will be open source. Our online multiplayer architecture is akin to the public server game infrastructure utilized by many games available via Steam, such as Minecraft or Counter-Strike.  As players join the server from their game client, the server owners begin earning our currency through the "DeathToll", or the minimal overhead fee charged by the blockchain for acting as a ledger for each logged game event, plus a configurable fee the server owner adds as one credit or "life" in the game (the "DeathToll"). Eventually, that price will be set by the market, as our community of gamers and server owners grow. The DeathToll pallet initially include's functionality to enable on chain interactions for first-person shooter game events (such as eliminating an enemy with a projectile attack), but will be expanded upon to include events for racing, sports, or tournament style fighting games. All data relevant to an individual's statistics in a game type will be permanently written on chain.  A leaderboard of the previous weekly, monthly, and yearly windows will be viewable via web3 browser. Eventually these wallet/browser menu's will be available in-game and through an iOS/Android wallet app, so that players can view their stats, accolades, and status. This will help drive engagement from the greater gaming community, who, as humans, yearn for everlasting glory.